node/sc/bridge_manager.go

```diff
@@ -304,48 +296,42 @@ func (bi *BridgeInfo) handleRequestValueTransferEvent(ev *RequestValueTransferEv
 
 	switch tokenType {
 	case KLAY:
-		handleTx, err = bi.bridge.HandleKLAYTransfer(auth, ev.Raw.TxHash, ev.From, ev.To, ev.ValueOrTokenId, ev.RequestNonce, ev.Raw.BlockNumber, ev.ExtraData)
+		handleTx, err = bi.bridge.HandleKLAYTransfer(auth, txHash, from, to, valueOrTokenId, requestNonce, blkNumber, extraData)
 		if err != nil {
 			return err
 		}
-		logger.Trace("Bridge succeeded to HandleKLAYTransfer", "nonce", ev.RequestNonce, "tx", handleTx.Hash().String())
-
+		logger.Trace("Bridge succeeded to HandleKLAYTransfer", "nonce", requestNonce, "tx", handleTx.Hash().String())
 	case ERC20:
-		handleTx, err = bi.bridge.HandleERC20Transfer(auth, ev.Raw.TxHash, ev.From, ev.To, tokenAddr, ev.ValueOrTokenId, ev.RequestNonce, ev.Raw.BlockNumber, ev.ExtraData)
+		handleTx, err = bi.bridge.HandleERC20Transfer(auth, txHash, from, to, ctpartTokenAddr, valueOrTokenId, requestNonce, blkNumber, extraData)
 		if err != nil {
 			return err
 		}
-		logger.Trace("Bridge succeeded to HandleERC20Transfer", "nonce", ev.RequestNonce, "tx", handleTx.Hash().String())
+		logger.Trace("Bridge succeeded to HandleERC20Transfer", "nonce", requestNonce, "tx", handleTx.Hash().String())
 	case ERC721:
-		// get URI of the ERC721
-		var uri string
-		erc721, err := scnft.NewERC721Metadata(ev.TokenAddress, bi.counterpartBackend)
-		if err != nil {
-			return err
-		}
-
-		uri, err = erc721.TokenURI(nil, ev.ValueOrTokenId)
-		if err != nil {
-			if err.Error() == vm.ErrExecutionReverted.Error() {
-				logger.Debug("Unable to get an ERC721 URI", "erc721", ev.TokenAddress.String(), "onParent", bi.onChildChain, "tokenId", ev.ValueOrTokenId.String())
-			} else {
-				return err
+		switch evType := ev.(type) {
+		case RequestValueTransferEvent:
```

Comment:
@hyunsooda Can we also make this refactored using interface?

Reply:
I was going to do that, but others held me back to do that. That's why I reverted my initial things and changed to this PR. Could you guys @henry-will @hqjang-pepper explain?

Reply:
@hyunsooda @hqjang-pepper 
May I suggest my opinion with my code. 

https://github.com/henry-will/klaytn/commit/79d70c80a8df2b7ad297caaef606ccf2835fc139

Reply:
Please check to compile first. It's not compiled, even failed to run the test. Some of the suggestion for variable naming seems good to me. 

Please check below:
1. Check compilation and test pass.
   - Moreover, you need to test the compatibility issue I mentioned in the PR description. This is the top-most important thing to us. It's behind the test of this PR.
3. Understand how Ethereum implemented event watcher and channel pipeline for solidity binding. 
FYI, the reason why I created two event channels is because of type enforcement [here](https://github.com/klaytn/klaytn/blob/dev/event/feed.go#L146). 
4. Why did you negelect `encodingVer`? Without this, we have no idea how we should decode correctly.
5. Refrain from making a `errors.new(...)` dynamically.
6. Two interface functions `GetVersion()` and `GetUri()` are not implemented for `RequestValueTransferEvent` struct. [It's wrong](https://github.com/henry-will/klaytn/commit/79d70c80a8df2b7ad297caaef606ccf2835fc139#diff-187d76a03ba63a723bebb7fc7901a8d74ac492124c0a8ebb73ae3404c6feae67R312). It's better to make an empty function to reduce verbose type assertion.

Reply:
I am sorry for compile error. It is just for share my opinion to you, because it is hard to explain about difference IRequestValueTransferEvent interface from RequestValueTransferEncodedEvent struct.

In "case ERC721:", If RequestValueTransferEvent transfer is empty uri "", I think that it isn't good response. And such case does not happen. Is It for netmarble case?

```go
		case RequestValueTransferEvent:
			handleTx, err = bi.bridge.HandleERC721Transfer(auth, txHash, from, to, ctpartTokenAddr, valueOrTokenId, requestNonce, blkNumber, "", extraData)
```
EncodingVer is only 2 currently. When other version add later, we must change some code. It is possible to decode now. I think that ERC721 should have uri, but your code skip or pass uri with "".

I don't know when `RequestValueTransferEvent` need version and uri. I think that RequestValueTransferEncodedEvent need version and uri only.

Reply:
I added detailed comments in your commit. Please check it out. Before you refactor the interface, let me emphasize the important stuff again.

1. The code must be compilable and pass the test.
2. For the compatibility test, clone [this repo](https://github.com/klaytn/servicechain-value-transfer-examples) first and run ERC-20, ERC-721, and KLAY examples. The `build` directory contains the old contract's abi and bin. You must replace them with the uploaded ones in this PR because the old contract does not have a new event emit. The repository does not have a bidirectional value transfer example. There's only one direction from parent chain to child chain for the ERC-20 and ERC-721. You have to make another value transfer test from the child chain to the parent chain. Please leave it if you have a question for testing.

Reply:
It committed at [79284e5](https://github.com/klaytn/klaytn/pull/1321/commits/79284e5cd168a71f4eb4550b3b598fb3630d2d55)